GoogleDataTransport/GDTCORLibrary/GDTCOREvent.m

```diff
@@ -74,8 +75,16 @@ - (void)setDataObject:(id<GDTCOREventDataObject>)dataObject {
   }
 }
 
-- (GDTCORStoredEvent *)storedEventWithDataFuture:(GDTCORDataFuture *)dataFuture {
-  return [[GDTCORStoredEvent alloc] initWithEvent:self dataFuture:dataFuture];
+- (void)setFileURL:(NSURL *)fileURL {
+  if (fileURL != _fileURL) {
+    _fileURL = fileURL;
+  }
+}
+
+- (void)clearDataObjectTransportBytes {
```

Comment:
I'd replace this with a method: `-writeToURL:` that calls [dataObject transportBytes] and writes the bytes to disk. Doing this allows you to get rid of the dataObjectTransportBytes property altogether. In GDTCORStorage, you'd create a file URL and then just call this method on the event.